### PR TITLE
CRM-19811 fix one instance of referring to LOWER() & comment others.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
I have tested & removed the current_employer reference & added notes to other references to LOWER. I will track down & remove the one relating to credit notes in the next round

---

 * [CRM-19811: Slow queries due to use of LOWER\(\) in mysql searches](https://issues.civicrm.org/jira/browse/CRM-19811)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I have only changed the 2 instances of LOWER for which I have written unit tests and commented all the rest. I am pretty confident in my analysis that they can all go but have limited myself to commenting the others for now so they can be removed later

File: CRM/Contribute/BAO/Query.php [hunk 1]
```diff
@@ -611,10 +611,10 @@ public static function whereClauseSingle(&$values, &$query) {
           $dataType = CRM_Utils_Type::typeToString($whereTable['type']);
         }
 
-        $wc = ($op != 'LIKE' && $dataType != 'Date') ? "LOWER($whereTable[where])" : "$whereTable[where]";
+        $wc = $whereTable['where'];
         $query->_where[$grouping][] = CRM_Contact_BAO_Query::buildClause($wc, $op, $value, $dataType);
         $query->_qill[$grouping][] = "$whereTable[title] $op $quoteValue";
-        list($tableName, $fieldName) = explode('.', $whereTable['where'], 2);
+        list($tableName) = explode('.', $whereTable['where'], 2);
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Query.php]: $fieldName is not used anywhere - minor tidy up

**Comment by eileenmcnaughton** [comment]
I can offer some quid-pro-quo review on this if anyone wants to review it

**Comment by ErichBSchulz** [comment]
this looks sound to me (for what that is worth!) - is there some kind of check in the installation code or admin panel that verifies that mysql is configured to case-insensitive text match?

something like `SELECT if('ok' = "OK", 'we are good', 'warning you need to adjust your MySQL configuration to do case-insensitive searching or things will break')`

**Comment by eileenmcnaughton** [comment]
Well the installation code specifies the table creation & in that it specifics the Collation - any collation ending in _ci is case insensitive. So, the only way I think someone could be other than case insensitive is if they have deliberately changed their Collation from the default - and I don't think we have to be responsible for every possible hack someone may have done

**Comment by eileenmcnaughton** [comment]
(ie. collation is set per table / field)

**Comment by ErichBSchulz** [comment]
oh. right. well that takes care of the only gotcha I could think of :-)

only other thought was is there a convention for tagging comments with `FIXME` ? vim highlights my FIXMEs and TODOs but I'm not sure if that is a thing...

**Comment by eileenmcnaughton** [comment]
We follow the drupal coding standards for those things

https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards#todo

**Comment by ErichBSchulz** [comment]
cool - no fixme then. worth a todo? (btw, I'm still wrestling with getting my proper tools operational but hopefully I'll be in a position to help expand unit test coverage soonish)

**Comment by colemanw** [comment]
@eileenmcnaughton I'll take you up on your quid-pro-quo offer... how about this one while we're on the topic of efficiency? #9623

**Pull request merged**